### DISCUSSION TITLE
Problem: I can't get `skip checkup` to pass on macOS 15 beta 1

### DISCUSSION THREAD

**Discussion by guitaripod** [skiptools/skip#146] [Q&A]
I have Android Studio installed. I've installed the Android SDK, but this checkup command fails like this

```shell
MaxBook ~ $ skip checkup
[✓] Skip version 0.8.50 (= 0.8.50)
[✓] macOS version 15.0 (> 13.5.0)
[✓] Swift version 5.10 (> 5.9.0)
[✓] Xcode version 15.4 (> 15.0.0)
[✓] Xcode tools SDKs: 6
[✓] Homebrew version 4.3.5 (> 4.1.0)
[✓] Gradle version 8.8 (> 8.6.0)
[✓] Java version 22.0.1 (> 21.0.0)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[⠙] Android Studio licenses[✓] Android Studio version: 2024.1
[✓] Android tools SDKs: 2
[✓] Resolve dependencies (6.91s)
[✓] Build hello-skip (10.57s)
[✓] Test Swift (6.89s)
[✗] Test Kotlin (5.05s)
[✓] Archive iOS ipa (15.12s)
[✓] Assemble HelloSkip-release.ipa 37 KB
[✓] Verify HelloSkip-release.ipa 37 KB
[✗] Assembling Android apk (12.91s)
[✗] Verify HelloSkip-release.apk: The file “app-release-unsigned.apk” couldn’t be opened because there is no such file.
[✓] Check Swift Package (0.2s)
[✗] Skip 0.8.50 checkup failed with 3 errors
Output logged to: /tmp/skip-checkup-2024-06-15T22:17:58Z.txt
See https://skip.tools/docs/faq and https://community.skip.tools for help
Error: 4 errors
```

![image](https://github.com/skiptools/skip/assets/47460844/e2f84a79-c8b9-4438-b633-006d9f65324a)
![image](https://github.com/skiptools/skip/assets/47460844/0878ccc1-0bec-44e5-89dd-40a00f48f418)

**Comment by guitaripod** [comment]
Here's the verbose output (included what i think is relevant):

```shell
BUILD FAILED in 12s
5 actionable tasks: 5 executed
[✗] Assembling Android apk (12.36s)
[✗] Verify HelloSkip-release.apk: The file “app-release-unsigned.apk” couldn’t be opened because there is no such file.
executing command: /usr/bin/swift package dump-package --package-path /var/folders/pm/l59hqgc51wj6wb6j7vynjfwc0000gn/T/B34A3A36-A91E-48D5-A387-4DB368892AE6/hello-skip
{
  "cLanguageStandard" : null,
  "cxxLanguageStandard" : null,
  "dependencies" : [
    {
      "sourceControl" : [
        {
          "identity" : "skip",
          "location" : {
            "remote" : [
              {
                "urlString" : "https://source.skip.tools/skip.git"
              }
            ]
          },
          "productFilter" : null,
          "requirement" : {
            "range" : [
              {
                "lowerBound" : "0.8.50",
                "upperBound" : "1.0.0"
              }
            ]
          }
        }
      ]
    },
    {
      "sourceControl" : [
        {
          "identity" : "skip-ui",
          "location" : {
            "remote" : [
              {
                "urlString" : "https://source.skip.tools/skip-ui.git"
              }
            ]
          },
          "productFilter" : null,
          "requirement" : {
            "range" : [
              {
                "lowerBound" : "0.0.0",
                "upperBound" : "1.0.0"
              }
            ]
          }
        }
      ]
    }
  ],
  "name" : "hello-skip",
  "packageKind" : {
    "root" : [
      "/private/var/folders/pm/l59hqgc51wj6wb6j7vynjfwc0000gn/T/B34A3A36-A91E-48D5-A387-4DB368892AE6/hello-skip"
    ]
  },
  "pkgConfig" : null,
  "platforms" : [
    {
      "options" : [
      ],
      "platformName" : "ios",
      "version" : "16.0"
    },
    {
      "options" : [
      ],
      "platformName" : "macos",
      "version" : "13.0"
    },
    {
      "options" : [
      ],
      "platformName" : "tvos",
      "version" : "16.0"
    },
    {
      "options" : [
      ],
      "platformName" : "watchos",
      "version" : "9.0"
    },
    {
      "options" : [
      ],
      "platformName" : "maccatalyst",
      "version" : "16.0"
    }
  ],
  "products" : [
    {
      "name" : "HelloSkipApp",
      "settings" : [
      ],
      "targets" : [
        "HelloSkip"
      ],
      "type" : {
        "library" : [
          "dynamic"
        ]
      }
    }
  ],
  "providers" : null,
  "swiftLanguageVersions" : null,
  "targets" : [
    {
      "dependencies" : [
        {
          "product" : [
            "SkipUI",
            "skip-ui",
            null,
            null
          ]
        }
      ],
      "exclude" : [
      ],
      "name" : "HelloSkip",
      "packageAccess" : true,
      "pluginUsages" : [
        {
          "plugin" : [
            "skipstone",
            "skip"
          ]
        }
      ],
      "resources" : [
        {
          "path" : "Resources",
          "rule" : {
            "process" : {
            }
          }
        }
      ],
      "settings" : [
      ],
      "type" : "regular"
    },
    {
      "dependencies" : [
        {
          "byName" : [
            "HelloSkip",
            null
          ]
        },
        {
          "product" : [
            "SkipTest",
            "skip",
            null,
            null
          ]
        }
      ],
      "exclude" : [
      ],
      "name" : "HelloSkipTests",
      "packageAccess" : true,
      "pluginUsages" : [
        {
          "plugin" : [
            "skipstone",
            "skip"
          ]
        }
      ],
      "resources" : [
        {
          "path" : "Resources",
          "rule" : {
            "process" : {
            }
          }
        }
      ],
      "settings" : [
      ],
      "type" : "test"
    }
  ],
  "toolsVersion" : {
    "_version" : "5.9.0"
  }
}
[✓] Check Swift Package (0.2s)
[✗] Skip 0.8.50 checkup failed with 3 errors
Output logged to: /tmp/skip-checkup-2024-06-15T22:22:10Z.txt
See https://skip.tools/docs/faq and https://community.skip.tools for help
Error: 4 errors
```

**Reply by aabewhite** [reply]
Can you please send the entire output log as an attachment?

**Reply by guitaripod** [reply]
sure. I switched my swift toolchain to Xcode 16 as well. I had it as 15.4. 
[skip-checkup-2024-06-15T22:34:27Z.txt.zip](https://github.com/user-attachments/files/15855529/skip-checkup-2024-06-15T22.34.27Z.txt.zip)

**Comment by aabewhite** [comment]
I see the following error on the log, which I haven’t seen before. How did you install Android Studio? Maybe open it up and see if it offers updates:Warning: SDK processing. This version only understands SDK XML versions up to 3 but an SDK XML file of version 4 was encountered. This can happen if you use versions of Android Studio and the command-line tools that were released at different times.Checking the license for package Android SDK Platform 34 in /Users/marcusziade/Library/Android/sdk/licensesLicense for package Android SDK Platform 34 accepted.Preparing "Install Android SDK Platform 34 (revision 3)".Warning: Failed to read or create install properties file.FAILURE: Build failed with an exception.* What went wrong:Could not determine the dependencies of task ':app:lintVitalReportRelease'.> Failed to install the following SDK components:      platforms;android-34 Android SDK Platform 34  Install the missing components using the SDK manager in Android Studio.Sent from my iPhoneOn Jun 15, 2024, at 5:37 PM, Marcus Ziadé ***@***.***> wrote:﻿
sure. I switched my swift toolchain to Xcode 16 as well. I had it as 15.4.
skip-checkup-2024-06-15T22:34:27Z.txt.zip

**Reply by guitaripod** [reply]
I installed it with brew as the `skip checkup` suggested

I'm seeing no error notifications at least: 
![CleanShot 2024-06-15 at 17 40 29](https://github.com/skiptools/skip/assets/47460844/721c9805-c686-441f-8b5c-4a72ae366430)

**Reply by aabewhite** [reply]
Did you maybe not have consistent internet access? I wonder if something failed to download. We've found that with our without Skip, getting a machine setup for Android development can sometimes be a pain.

I'm going to go off of the error message, which says "Failed to install the following SDK components:      platforms;android-34 Android SDK Platform 34  Install the missing components using the SDK manager in Android Studio"

In Android Studio, select Tools -> SDK Manager. Try installing Android 14, which is API Level 34. Let's see if that does it.

I'm not our tools expert, but unfortunately he's going out of the country tomorrow and is probably busy with that.

**Reply by guitaripod** [reply]
I have Platform 34 installed. My screenshots above show what I have installed. don't think i need to have the entire section installed, right?

**Reply by aabewhite** [reply]
Sorry I missed that. 

This is very strange. Can you please try also installing Android 13, then re-launching Android Studio and trying again?

I mean I guess this could be an OS beta issue, but seems unlikely.

**Reply by marcprux** [reply]
I've never seen that error either. I think @aabewhite's suggestion of installing Android 13 might help. 

Looking around the internet for the error:

```
Warning: SDK processing. This version only understands SDK XML versions up to 3 but an SDK XML file of version 4 was encountered. This can happen if you use versions of Android Studio and the command-line tools that were released at different times.
Checking the license for package Android SDK Platform 34 in /Users/marcusziade/Library/Android/sdk/licenses
License for package Android SDK Platform 34 accepted.
Preparing "Install Android SDK Platform 34 (revision 3)".
Warning: Failed to read or create install properties file.
FAILURE: Build failed with an exception.
* What went wrong:
Could not determine the dependencies of task ':app:lintVitalReportRelease'.
> Failed to install the following SDK components:
      platforms;android-34 Android SDK Platform 34
  Install the missing components using the SDK manager in Android Studio.
```

I've found a few references like [this](https://github.com/bitrise-steplib/steps-install-missing-android-tools/issues/57) and [this](https://stackoverflow.com/questions/69162605/jenkins-build-for-android-sdk-problem-failed-to-install-the-following-sdk-compon) that suggest it might be a permissions problem. Can you check to see if you have write permission to `/Users/marcusziade/Library/Android/sdk/`? You might also try installing the tools package with `brew reinstall android-platform-tools`.

Also, I would expect that this issue would affect any attempt to build an Android project. Could you try creating a quick new Android project in Android Studio and seeing if it builds? It might have a clearer error message if it is also failing.

**Reply by guitaripod** [reply]
- I have write access to the directory where the SDK is 
```shell
MaxBook ~/Library/Android/sdk $ ls -le
total 0
drwxr-xr-x   3 marcusziade  staff   96 Jun 15 14:46 build-tools
drwxr-xr-x  29 marcusziade  staff  928 Jun 15 14:46 emulator
drwxr-xr-x   4 marcusziade  staff  128 Jun 15 14:34 licenses
drwxr-xr-x  15 marcusziade  staff  480 Jun 15 14:46 platform-tools
drwxr-xr-x   3 marcusziade  staff   96 Jun 15 14:35 platforms
drwxr-xr-x   3 marcusziade  staff   96 Jun 15 14:46 sources
drwxr-xr-x   3 marcusziade  staff   96 Jun 15 14:34 system-images
```

- A template project builds successfully
![CleanShot 2024-06-16 at 17 57 26](https://github.com/skiptools/skip/assets/47460844/186b20f5-0fe3-444b-8410-f878609df14d)

- Installed the additional `android-platform-tools`
- I installed the 13 SDK and platform as well


Still fails the same :*(

**Reply by marcprux** [reply]
We'll figure this out.

Can you try building that project from the Terminal and see if that makes a difference? There might be some difference in behavior when building from the command line. E.g.,

```console
cd /Users/marcusziade/ AndroidStudioProjects/MyApplication
gradle assembleDebug
```

Also, can you show us the output of the `env` command?

**Reply by aabewhite** [reply]
Another question is: do you have access to another machine that isn't running the beta? It would be very odd if this were a beta issue... but it's also odd that we haven't seen this problem before. Definitely follow @marcprux's advice to debug, but if you just want to move forward and don't care if it's on a different machine, you might try it.

**Reply by guitaripod** [reply]
Thanks, I'll get back to this during the next break from work
